splinterd/tests/framework/builders.rs

```diff
@@ -0,0 +1,966 @@
+// Copyright 2018-2021 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Provides functionality for building [`CircuitBuilder`]s, used in the admin service
+//! integration tests.
+
+use splinter::admin::messages::{
+    AuthorizationType, BuilderError, CircuitStatus, CreateCircuitBuilder, DurabilityType,
+    PersistenceType, RouteType, SplinterNode, SplinterNodeBuilder, SplinterService,
+    SplinterServiceBuilder,
+};
+
+use crate::admin::payload::{
+    complete_create_payload, make_circuit_proposal_vote_payload, ManagementPayloadError,
+};
+use crate::framework::network::Network;
+use cylinder::SigningError;
+use splinter::admin::client::event::{
+    AdminServiceEvent, AdminServiceEventClient, EventType, NextEventError,
+};
+use splinter::admin::error::MarshallingError;
+use splinter::error::{InternalError, InvalidArgumentError};
+use splinterd::node::Node;
+use std::error::Error;
+use std::fmt;
+
+/// `CircuitBuildError` represents any errors that may arise during the final circuit build step
```

Comment:
Once again, this could become a full directory `builder/error.rs`, and then the rest of the builders defined within that directory. Seems like there are a lot of separate builders here